cmd/gaia/app/x/bank/handler.go

```diff
@@ -0,0 +1,100 @@
+// Package bank contains a forked version of the bank module. It only contains
+// a modified message router to support a very limited form of transfers during
+// mainnet launch -- MsgMultiSend messages.
+//
+// NOTE: This fork should be removed entirely once transfers are enabled and
+// the Gaia router should be reset to using the original bank module router.
+package bank
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+)
+
+var (
+	uatomDenom    = "uatom"
+	atomsToUatoms = int64(1000000)
+
+	// BurnedCoinsAccAddr represents the burn account address used for
+	// MsgMultiSend message during the period for which transfers are disabled.
+	// Its Bech32 address is cosmos1x4p90uuy63fqzsheamn48vq88q3eusykf0a69v.
+	BurnedCoinsAccAddr = sdk.AccAddress(crypto.AddressHash([]byte("bankBurnedCoins")))
+)
+
+// NewHandler returns a handler for "bank" type messages.
+func NewHandler(k bank.Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		switch msg := msg.(type) {
+		case bank.MsgSend:
+			return handleMsgSend(ctx, k, msg)
+
+		case bank.MsgMultiSend:
+			return handleMsgMultiSend(ctx, k, msg)
+
+		default:
+			errMsg := "Unrecognized bank Msg type: %s" + msg.Type()
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+// handleMsgSend implements a MsgSend message handler. It operates no differently
+// than the standard bank module MsgSend message handler in that it transfers
+// an amount from one account to another under the condition of transfers being
+// enabled.
+func handleMsgSend(ctx sdk.Context, k bank.Keeper, msg bank.MsgSend) sdk.Result {
```

Comment:
(if we remove the genesis parameter we can get rid of this)

Reply:
done